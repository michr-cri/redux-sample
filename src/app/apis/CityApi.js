import 'jquery';
//window.$ = window.jQuery = jQuery;

class CityApi {
    static fetchCities(page, pageSize) {
        let url = APP_URL + '/cities';
        if(page && pageSize) {
            url += '?page=' + page + '&page-size=' + pageSize;
        }
        return jQuery.ajax({
            method: 'GET',
            url: url,
            header: {
                'Content-Type': 'application/json'
            }
        });
    }

}

export default CityApi;