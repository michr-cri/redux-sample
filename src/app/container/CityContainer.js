import { connect } from 'react-redux';
import City from '../components/City';
import {fetchCities, goToPage} from '../actions/action.city';

function mapStateToProps(state) {
    console.log(state.city.cities);
    return {
        cities: state.city.cities,
        totalCount: state.city.totalCount,
        page: state.city.page,
        pageSize: state.city.pageSize
    }
}

function mapDispatchToProps(dispatch) {
    return {
        fetchCities: (page, pageSize) => {
            dispatch(fetchCities(page, pageSize));
        },
        goToPage: (page) => {
            dispatch(goToPage(page));
        }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(City);