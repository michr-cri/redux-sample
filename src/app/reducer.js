import { combineReducers } from 'redux'
import typeioReducer from './state/reducers/reducer.typeio';
import feedbackReducer from './shared/reducers/reducer.feedback';
import stateReducer from './state/reducers/reducer.state';
import loginReducer from './login/reducers/reducer.login';
import cityReducer from './city/reducers/reducer.city';
import {reducer as reduxFormReducer} from 'redux-form';

export default combineReducers({
    state: stateReducer,
    typeio: typeioReducer,
    feedback: feedbackReducer,
    login: loginReducer,
    city: cityReducer,
    form: reduxFormReducer
});