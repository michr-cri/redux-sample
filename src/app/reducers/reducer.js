import { combineReducers } from 'redux'
import typeioReducer from './reducer.typeio';
import feedbackReducer from './reducer.feedback';
import appReducer from './reducer.app';
import loginReducer from './reducer.login';

export default combineReducers({
    app: appReducer,
    typeio: typeioReducer,
    feedback: feedbackReducer,
    login: loginReducer
});