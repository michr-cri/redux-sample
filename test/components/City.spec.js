import React from 'react';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import reducer from '../../src/app/reducers/reducer';
import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import {findDOMNode} from 'react-dom';
import { Provider } from 'react-redux';
import {Router} from 'react-router-dom';
import history from '../../src/app/shared/history';
import CityApi from '../../src/app/apis/CityApi';
import CityContainer from '../../src/app/container/CityContainer';

import jQuery from 'jquery';
window.$ = window.jQuery = jQuery;

let component;
describe('components', () => {
    describe('City', () => {

        beforeAll(()=> {
            let initialState = {
                city: {
                    page: 1,
                    pageSize: 2,
                    cities: [],
                    totalCount: 0
                }
            };
            let cities = [{id: 1, name: 'Ann Arbor'}, {id: 2, name: 'Detroit'}];
            let totalCount = 7;
            let paginationResult = {page: cities, totalCount: totalCount};
            let fetchCitiesPromise = new Promise(function(resolve) {resolve(paginationResult);});

            jest.spyOn(CityApi, 'fetchCities').mockReturnValueOnce(fetchCitiesPromise);
            let store = createStore(reducer, initialState, applyMiddleware(thunk));

            component  = ReactTestUtils.renderIntoDocument(
                <Provider store={store}>
                    <Router history={history}>
                        <CityContainer />
                    </Router>
                </Provider>
            );
        });

        it('should render self', () => {
            let componentDOM = findDOMNode(component);
            expect(componentDOM.querySelectorAll('.city-list>li').length).toEqual(2);

            expect(componentDOM.querySelectorAll('.city-list>li')[0].innerHTML).toContain('Ann Arbor');
            expect(componentDOM.querySelectorAll('.city-list>li')[1].innerHTML).toContain('Detroit');
        });

        it('Prev link clicked', () => {
            let cities = [{id: 3, name: 'New York'}, {id: 4, name: 'LA'}];
            let totalCount = 7;
            let paginationResult = {page: cities, totalCount: totalCount};
            let fetchCitiesPromise = new Promise(function(resolve) {resolve(paginationResult);});

            jest.spyOn(CityApi, 'fetchCities').mockReturnValueOnce(fetchCitiesPromise);
            let componentDOM = findDOMNode(component);
            let prevLink = componentDOM.querySelector('#aPrev');
            console.log('before click');
            ReactTestUtils.Simulate.click(prevLink);
            console.log('after click');

            component.forceUpdate();

            let newComponentDOM = findDOMNode(component);

            console.log(newComponentDOM.querySelectorAll('.city-list>li')[0]);
            expect(newComponentDOM.querySelectorAll('.city-list>li').length).toEqual(2);

            expect(newComponentDOM.querySelectorAll('.city-list>li')[0].innerHTML).toContain('New York');
            expect(newComponentDOM.querySelectorAll('.city-list>li')[1].innerHTML).toContain('LA');
        });
    })
});

