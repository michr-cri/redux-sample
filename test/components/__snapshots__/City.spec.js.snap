// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components City should render self 1`] = `
<div>
  <ul
    className="city-list"
  >
    <li>
      Ann Arbor
    </li>
    <li>
      Detroit
    </li>
  </ul>
  <a
    className="disabled"
    href="javascript:void(0);"
    id="aPrev"
    onClick={[Function]}
  >
    Prev
  </a>
  <a
    href="#/cities?page=1&page-size=2"
    onClick={[Function]}
  >
    1
  </a>
  <a
    href="#/cities?page=2&page-size=2"
    onClick={[Function]}
  >
    2
  </a>
  <a
    href="#/cities?page=3&page-size=2"
    onClick={[Function]}
  >
    3
  </a>
  <a
    href="#/cities?page=4&page-size=2"
    onClick={[Function]}
  >
    4
  </a>
  <a
    className=""
    href="javascript:void(0);"
    onClick={[Function]}
  >
    Next
  </a>
</div>
`;
